src/code42cli/click_ext/types.py

```diff
@@ -116,3 +116,19 @@ def _get_dt_from_date_time_pair(date, time):
             raise BadParameter("Unable to parse date string: {}.".format(date_string))
         else:
             return dt
+
+
+class MapChoice(click.Choice):
+    """Choice subclass that takes an extra map of additional 'valid' keys to map to correct
+    choices list, allowing backward compatible choice changes. The extra keys don't show up
```

Comment:
Would it make more sense to have the extra keys be the ones that show up in the help text and to just have this be a way to make things backwards compatible more silently?

I guess I could go either way, just curious if you thought about that

Reply:
I did, but if a new category gets added, it will _only_ be camelcase in py42 going forward, so we'd have to keep adding new upper-case maps as well to keep it consistent.

Reply:
agree with @timabrmsn , we do not have to support both casings in newer categories that never used the old casing.